garrysmod/lua/includes/extensions/player.lua

```diff
@@ -52,7 +52,43 @@ function player.GetByUniqueID( ID )
 		if ( IsValid( pl ) && pl:IsPlayer() && pl:UniqueID() == ID )	then
 			return pl
 		end
-		
+
+	end
+
+	return false
+
+end
+
+function player.GetBySteamID( ID )
+
+	ID = string.upper( ID )
+
+	assert( string.find( ID, "STEAM_" ), "player.GetBySteamID: passed string '" .. ID .. "' is not a valid steamid!" )

```

Comment:
I recommend you use ID:StartWith("STEAM_") here; it's more relevant to SteamIDs, and it's also considerably quicker.